pixie/stdlib.pxi

```diff
@@ -1304,9 +1312,13 @@ and implements IAssociative, ILookup and IObject."
                                           fields)))
                         `(-hash [self]
                                 (hash [~@field-syms]))]
-        deftype-decl `(deftype ~nm ~fields ~@default-bodies ~@body)]
+        deftype-decl `(deftype ~nm ~(conj fields '__meta) ~@default-bodies ~@body)
+        ctor `(defn ~ctor-name ~field-syms
```

Comment:
Is this separate constructor necessary?  The `deftype` call already defines it, so we would overwrite that here...